### PR TITLE
Codemon reg validation

### LINKED ISSUE #63: validation on codemon reg
When using codemon reg, empty username and password are also stored to the meta.json file.
We should use a validation here.
Also, If there is already a meta.json file, then, it overwrites the meta.json file. 
We also need a validation here.

### PR DESCRIPTION
**Pull request by lucifer0987**
closes #63 
Added validations for user input on the meta.
Also added a check before overwriting the meta file.

### CODE REVIEW

File: codemon/CodemonReg.py [hunk 1]
```diff
@@ -17,22 +17,47 @@ def codemonReg():
...
     dataDict["name"] = name
     dataDict["codeforces_username"] = codeforces_username
     dataDict["codeforces_password"] = codeforces_password
     dataDict["github_username"] = github_username
     dataDict["github_password"] = github_password
     dataJson = json.dumps(dataDict, indent=4)
-    full_filename = os.path.join(home, file_name, "meta.json")
+    full_filename = os.path.join(path, "meta.json")
     with open(full_filename, 'w+') as f:
         f.write(dataJson)
```
Comment (ankushbhardwxj) [inline codemon/CodemonReg.py:64]: Can you print the dataJson here to give preview of what was written to meta.json ? Also add a message at the end which says, metafile updated successfully.

File: codemon/CodemonReg.py [hunk 2]
```diff
@@ -17,22 +17,47 @@ def codemonReg():
         if os.name == 'nt':
             FILE_ATTRIBUTE_HIDDEN = 0x02
             ret = ctypes.windll.kernel32.SetFileAttributesW(path, FILE_ATTRIBUTE_HIDDEN)
+        writeMeta(path)
+    else:
+        if os.path.exists(os.path.join(path, "meta.json")):
+            print("Meta file already exists.")
```
Comment (ankushbhardwxj) [inline codemon/CodemonReg.py]: Rewrite "Meta file" as "Metafile"

File: codemon/CodemonReg.py [hunk 3]
```diff
@@ -17,22 +17,47 @@ def codemonReg():
...
+        print("No Input Detected")
+        codeforces_password = stdiomask.getpass(prompt="Codeforces password : ")
+    github_username = input("\nGithub username (without @) : ")
+    while (github_username == ''):
+        print("No Input Detected")
+        github_username = input("Github username (without @) : ")
+    github_password = stdiomask.getpass(prompt="\nGitHub Password : ")
+    while (github_password == ''):
+        print("No Input Detected")
+        github_password = stdiomask.getpass(prompt="GitHub Password : ")
```
Comment (ankushbhardwxj) [inline codemon/CodemonReg.py]: Please get rid of the newlines inside of the `print()` here, it causes inconsistent spacing on terminal.

File: codemon/CodemonReg.py [hunk 4]
```diff
@@ -17,22 +17,47 @@ def codemonReg():
...
     print("\nPlease enter the following data.\nYour sensitive information is collected for making your life easier while using Codemon.\nIt will be stored on your local machine.\nIt won't be stored or used by Codemon developers for any purpose. ")
     name = input("\nEnter Name : ")
-    codeforces_username = input("Codeforces username (without @) : ")
-    codeforces_password = stdiomask.getpass(prompt="Codeforces password : ")
-    github_username = input("Github username (without @) : ")
-    github_password = stdiomask.getpass(prompt="GitHub Password : ")
+    while (name == ''):
+        print("No Input Detected")
+        name = input("Enter Name : ")
+    codeforces_username = input("\nCodeforces username (without @) : ")
```
Comment (ankushbhardwxj) [inline codemon/CodemonReg.py]: Do you think if we remove "`(without @)`" here, and write it as `Codeforces username: @ <here user will input>` will look better and more professional ?

**Review by ankushbhardwxj — CHANGES_REQUESTED** [review]

File: codemon/CodemonReg.py (hunk 4 above)
Comment (lucifer0987) [reply codemon/CodemonReg.py]: @ankingcodes I guess, it will be more confusing.
"(without @)" tells them to specifically enter the username without "@"

**Comment by lucifer0987** [comment]
@ankingcodes made changes. Please review.

**Comment by ankushbhardwxj** [comment]
@lucifer0987 - Looks good ! Thanks.

**Pull request merged**